### DISCUSSION TITLE
hass-emulated-hue and deconz: Dimmable lights issue

### DISCUSSION THREAD

**Discussion by mullermn** [hass-emulated-hue/core#253] [Q&A]
Hi,

I haven't logged this as a bug because it's possible that the problem is not with emulated_hue - I'm just wondering if anyone has any ideas, please.

My configuration looks like this:
[Hue bulbs] -> [conbee 2/deconz] -> [home assistant] -> [hass-emulated-hue] -> [Alexa]

I'm having an issue getting dimmable bulbs exposed by deconz to present to Alexa. The bulbs show up as lights in home assistant and work correctly there, and they are exposed on the /api endpoint by emulated_hue with what looks like a perfectly valid configuration, however they're not being found by Alexa.

Alexa DOES correctly pick up lights derived from switches and light groups exposed by home assistant.

I believe you can use emulated_hue in parallel with deconz and have both picked up by Alexa, but I don't want to do this because a) deconz and emulated_hue are running in containers on the same host and can't both have port 80, and b) I want to be able to wrap non-bulbs up in the same groups etc that Alexa sees.

An example of a working and detected light (this is really a mains switch presented as a light by HA):
```
    "16": {
      "state": {
        "on": false,
        "reachable": true,
        "mode": "homeautomation"
      },
      "name": "Barn security light",
      "uniqueid": "00:7a:7e:bd:2c:ac:6f:0f-3d",
      "swupdate": {
        "state": "noupdates",
        "lastinstall": "2021-12-27T08:56:20"
      },
      "config": {
        "archetype": "sultanbulb",
        "direction": "omnidirectional",
        "function": "mixed",
        "startup": {
          "configured": true,
          "mode": "safety"
        }
      },
      "capabilities": {},
      "type": "On/off light",
      "manufacturername": "Signify Netherlands B.V.",
      "productname": "On/off light",
      "modelid": "LTW010",
      "swversion": "5.127.1.26581"
    },
```

And an example of a genuine hue bulb presented by deconz to HA and from HA to emulated_hue but NOT picked up by Alexa:
```    "7": {
      "state": {
        "on": true,
        "reachable": true,
        "mode": "homeautomation",
        "bri": 254
      },
      "name": "Sitting room light",
      "uniqueid": "00:17:88:01:02:fe:b5:62",
      "swupdate": {
        "state": "noupdates",
        "lastinstall": "2021-12-27T08:56:20"
      },
      "config": {
        "archetype": "sultanbulb",
        "direction": "omnidirectional",
        "function": "mixed",
        "startup": {
          "configured": true,
          "mode": "safety"
        }
      },
      "type": "Dimmable light",
      "capabilities": {
        "certified": true,
        "control": {
          "maxlumen": 840,
          "mindimlevel": 5000
        },
        "streaming": {
          "proxy": false,
          "renderer": false
        }
      },
      "manufacturername": "Philips",
      "productname": "Sitting room light",
      "modelid": "LWB010",
      "swversion": "1.29.0_r21169"
    },    
```

This issue refers to the reverse problem (dimmable lights being detected but on/off lights not being picked up) and is for the original emulated_hue addon for HA, not hass-emulated-hue, but I wonder if the problem is in the same ballpark in that it's Alexa being very picky in the way it matches one of the attributes of the light?: 
https://github.com/home-assistant/core/issues/30013

Does anyone have any ideas, please? Failing that, what's the best way to override the config for individual lights so I can play around and see what has to change for Alexa to find it?

Thanks

**Comment by alexyao2015** [comment]
This indeed would be a bug with the emulated API. You'll need to access the actual API through the same endpoint that Alexa uses, which you can obtain by viewing the logs after enabling the debug flag. You will then need to view the code in api.py and edit the corresponding endpoint to return the correct data. 

If you would simply be able to compare the data from the home assistant endpoint and our endpoint, that should be enough information to fix the issue.
